src/app/app.component.ts

```diff
@@ -1185,7 +1119,10 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
     if (this.singlemailviewer.messageId) {
       fragment += `:${this.singlemailviewer.messageId}`;
     }
-    this.router.navigate(['/'], { fragment });
+    // don't update if already there, its a bit circular!
+    if (!this.router.url.match('#fragment$')) {
```

Comment:
Does this work? Looks like a weird string-regex hybrid.

Reply:
Erm no.. I've just removed it as it should be redundant, now that the fragment subscribe code only saves the value, instead of doing stuff with it (it was at one point going around in circles)